### PR TITLE
measurement-kit: add new package

### PR DESCRIPTION
**Pull request by ja-pa**
Maintainer: me / @ja-pa  
Compiled tested: cortexa53, Turris MOX, OpenWrt 18.06.1
Run tested: cortexa53, Turris MOX, OpenWrt 18.06.1

Description:

Measurement-kit is library and tool that implements open network measurement methodologies (performance, censorship). It's useful for collecting information about network.
More https://measurement-kit.github.io/

### CODE REVIEW

File: libs/libmaxminddb/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (C) 2019 OpenWrt.org
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Don't assign copyright to OpenWrt.org unless you're affliated. Read https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md

File: libs/libmaxminddb/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+PKG_MAINTAINER:=Jan Pavlinec <jan.pavlinec@nic.cz>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/libmaxminddb
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=A library for working with MaxMind DB files
+  URL:=https://maxmind.github.io/libmaxminddb/
+  #DEPENDS:=+libstdcpp
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Remove

File: libs/libmaxminddb/Makefile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	$(CP) $(PKG_INSTALL_DIR)/usr/include/* $(1)/usr/include/
+	$(INSTALL_DIR) $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmaxminddb.{a,so*} $(1)/usr/lib/
+endef
+
+define Package/libmaxminddb/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/mmdblookup $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/usr/lib
+
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Remove blank line

File: libs/libmaxminddb/Makefile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libmaxminddb
+PKG_VERSION:=1.3.2
+PKG_RELEASE=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/maxmind/libmaxminddb/releases/download/$(PKG_VERSION)/
+PKG_HASH:=e6f881aa6bd8cfa154a44d965450620df1f714c6dc9dd9971ad98f6e04f6c0f0
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Do you really need to define PKG_BUILD_DIR?

File: libs/measurement-kit/Makefile [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (C) 2019 OpenWrt.org
```
Comment (diizzyy) [inline libs/measurement-kit/Makefile]: Same as before

File: libs/measurement-kit/Makefile [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=measurement-kit
+PKG_VERSION:=0.9.1
+PKG_RELEASE=1
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline libs/measurement-kit/Makefile]: This will clash, use codeload instead

File: libs/measurement-kit/Makefile [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+PKG_SOURCE_URL:=https://github.com/measurement-kit/measurement-kit/archive/
+PKG_HASH:=5a4f0ae8c9ed83f186231e4b04f25f026b8a46f4171045ef972c3d9bdc912081
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (diizzyy) [inline libs/measurement-kit/Makefile]: Not needed?

File: libs/libmaxminddb/Makefile [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	$(INSTALL_DIR) $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmaxminddb.{a,so*} $(1)/usr/lib/
+endef
+
+define Package/libmaxminddb/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/mmdblookup $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/usr/lib
+
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmaxminddb.{a,so*} $(1)/usr/lib/
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Why is libmaxminddb.a needed?

File: libs/measurement-kit/Makefile [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+define Build/Configure
+	( cd $(PKG_BUILD_DIR); ./autogen.sh )
+	$(call Build/Configure/Default)
+endef
+
+define Package/measurement-kit/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/measurement_kit $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmeasurement_kit.{la,a} $(1)/usr/lib/
```
Comment (diizzyy) [inline libs/measurement-kit/Makefile]: .la and .a instead of so?

File: libs/libmaxminddb/Makefile (hunk 1 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: I replaced that with proper copyright. :man_facepalming:

File: libs/libmaxminddb/Makefile (hunk 2 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: Removed

File: libs/libmaxminddb/Makefile (hunk 3 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: Removed

File: libs/libmaxminddb/Makefile (hunk 4 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: PKG_BUILD_DIR not needed.

File: libs/measurement-kit/Makefile (hunk 5 above)
Comment (ja-pa) [reply libs/measurement-kit/Makefile]: Replaced  :man_facepalming:

File: libs/measurement-kit/Makefile (hunk 6 above)
Comment (ja-pa) [reply libs/measurement-kit/Makefile]: Replaced with codeload.

File: libs/measurement-kit/Makefile (hunk 7 above)
Comment (ja-pa) [reply libs/measurement-kit/Makefile]: PKG_BUILD_DIR not needed. Removed.

File: libs/libmaxminddb/Makefile (hunk 8 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: libmaxminddb.a not needed. Removed from install

File: libs/measurement-kit/Makefile (hunk 9 above)
Comment (ja-pa) [reply libs/measurement-kit/Makefile]: .so isn't builded right now and la/a isn't need on target right now. Removed from install.

**Comment by ja-pa** [comment]
@diizzyy I added suggested changes, what do you think now?

File: libs/libmaxminddb/Makefile [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/usr/include/
+	$(CP) $(PKG_INSTALL_DIR)/usr/include/* $(1)/usr/include/
+	$(INSTALL_DIR) $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmaxminddb.{a,so*} $(1)/usr/lib/
+endef
+
+define Package/libmaxminddb/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/mmdblookup $(1)/usr/bin/
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Use ```$(INSTALL_BIN)```

File: libs/libmaxminddb/Makefile [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+PKG_NAME:=libmaxminddb
+PKG_VERSION:=1.3.2
+PKG_RELEASE=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/maxmind/libmaxminddb/releases/download/$(PKG_VERSION)/
+PKG_HASH:=e6f881aa6bd8cfa154a44d965450620df1f714c6dc9dd9971ad98f6e04f6c0f0
+
+
```
Comment (diizzyy) [inline libs/libmaxminddb/Makefile]: Remove empty line

File: libs/measurement-kit/Makefile [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+CONFIGURE_ARGS+= --with-ca-bundle=/etc/ssl/cert.pem
+
+define Build/Configure
+	( cd $(PKG_BUILD_DIR); ./autogen.sh )
+	$(call Build/Configure/Default)
+endef
+
+define Package/measurement-kit/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/measurement_kit $(1)/usr/bin/
```
Comment (diizzyy) [inline libs/measurement-kit/Makefile]: Use ```$(INSTALL_BIN)```

File: libs/libmaxminddb/Makefile (hunk 10 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: fixed

File: libs/libmaxminddb/Makefile (hunk 11 above)
Comment (ja-pa) [reply libs/libmaxminddb/Makefile]: removed

File: libs/measurement-kit/Makefile (hunk 12 above)
Comment (ja-pa) [reply libs/measurement-kit/Makefile]: fixed

**Comment by ja-pa** [comment]
@diizzyy What do you think? Is it ready to merge?

**Comment by diizzyy** [comment]
Looks good to me, you should probably bump measurement-kit to current version however. Perhaps you can bring down the size of measurement-kit slightly by adding ```-flto```?

**Comment by ja-pa** [comment]
@diizzyy I updated measurement-kit version to 0.9.2 and it works ok. I also tried to use flto but it breaks build right now.

**Comment by diizzyy** [comment]
@ja-pa 
Great, except the flto doesn't but it's not the end of the world and may not reduce binary size in the end.

**Comment by ja-pa** [comment]
@hnyman I think it's ready to merge.

**Comment by hnyman** [comment]
This fails to compile for some targets.

So far buildbot has succeeded for two ( arm_cortex-a9_neon, armeb_xscale) and failed for two (mipsel_24kc_24kf, armeb_xscale)

https://downloads.openwrt.org/snapshots/faillogs/mipsel_24kc_24kf/packages/measurement-kit/compile.txt
https://downloads.openwrt.org/snapshots/faillogs/armeb_xscale/packages/measurement-kit/compile.txt

Some little-endian / big-endian thing?

If there is a target compatibility restriction, it should be defined in the Makefile so that it is not tried to be compiled for targets known to be unsupported.

**Pull request merged**